packages/drivers/odsp-driver/src/test/odspDriverUrlResolver2Tests.spec.ts

```diff
@@ -0,0 +1,142 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable max-len */
+/* eslint-disable dot-notation */
+
+import { strict as assert } from "assert";
+import sinon from "sinon";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { OdspDriverUrlResolver2 } from "../odspDriverUrlResolver2";
+import { SharingLinkScopeFor, SharingLinkTokenFetcher } from "../tokenFetch";
+import { getHashedDocumentId } from "../odspUtils";
+import { createOdspUrl } from "../createOdspUrl";
+import * as graph from "../graph";
+import { getLocatorFromOdspUrl, storeLocatorInOdspUrl } from "../odspFluidFileLink";
+import { IOdspResolvedUrl, SharingLinkHeader } from "../contracts";
+
+describe("OdspUrlResolver3", () => {
+    const siteUrl = "https://microsoft.sharepoint-df.com/siteUrl";
+    const driveId = "driveId";
+    const itemId = "fileId";
+    const dataStorePath = "dataStorePath";
+    const fileName = "fileName";
+    const sharelink = "https://microsoft.sharepoint-df.com/site/SHARELINK";
+    let urlResolver: OdspDriverUrlResolver2;
+
+    beforeEach(() => {
+        const shareLinkTokenFetcher: SharingLinkTokenFetcher =
+            async (siteURL: string, scopeFor: SharingLinkScopeFor, refresh: boolean) => "SharingLinkToken";
+            urlResolver = new OdspDriverUrlResolver2(shareLinkTokenFetcher);
```

Comment:
Any chance we can give it descriptive name instead of adding 2 to OdspDriverUrlResolver?

Reply:
So it is just another odsp driver resolver with share link handling/generation capability which is internal implementation detail. Maybe rename it to:  OdspDriverUrlResolverForShareLink but I think it is not necessary.

Reply:
Renamed